libpod/container_internal_linux.go

```diff
@@ -547,10 +547,8 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
 			}
 		}
 		if IP != nil {
-			env := fmt.Sprintf("IP=%s", IP)
 			// Tell CNI which IP address we want.
-			os.Setenv("CNI_ARGS", env)
-			logrus.Debugf("Restoring container with %s", env)
+			c.config.StaticIP = IP
```

Comment:
Generally, this is really unsafe - our code assumes `c.config` will never change once it's loaded into the database. maybe add a `requestedIP` field to the Container struct, and use that preferentially if set?

Reply:
Ah, good to know. Thanks. I will change it.